src/algebra/hom/ring.lean

```diff
@@ -391,6 +391,14 @@ protected lemma map_bit0 (f : α →+* β) : ∀ a, f (bit0 a) = bit0 (f a) := m
 /-- Ring homomorphisms preserve `bit1`. -/
 protected lemma map_bit1 (f : α →+* β) : ∀ a, f (bit1 a) = bit1 (f a) := map_bit1 f
 
+@[simp] lemma apply_ite_zero_one (f : α →+* β) (p : Prop) [decidable p] :
+  f (ite p 0 1) = ite p 0 1 :=
+by { split_ifs; simp [h] }
+
+@[simp] lemma apply_ite_one_zero (f : α →+* β) (p : Prop) [decidable p] :
```

Comment:
These are slightly weird lemmas because the proof is so trivial once we get `f` on the other side of the if. But on the other hand, I'm not sure that `simp` should always rewrite `f (ite p a b)` to `ite p (f a) (f b)`.

In any case, we can generalize this to any subtype of `ring_hom`:
```suggestion
@[simp] lemma map_ite_zero_one {F : Type*} [ring_hom_class F α β] (f : F) (p : Prop) [decidable p] :
  f (ite p 0 1) = ite p 0 1 :=
by { split_ifs; simp [h] }

@[simp] lemma map_ite_one_zero {F : Type*} [ring_hom_class F α β] (f : F) (p : Prop) [decidable p] :
```